core/src/main/kotlin/com/malinskiy/marathon/execution/DevicePoolActor.kt

```diff
@@ -74,8 +74,24 @@ class DevicePoolActor(private val poolId: DevicePoolId,
         queue.send(QueueMessage.Completed(device, results))
     }
 
-    private suspend fun deviceReady(msg: DevicePoolMessage.FromDevice.RequestNextBatch) {
-        queue.send(QueueMessage.RequestBatch(msg.device))
+    private suspend fun deviceReady(msg: DevicePoolMessage.FromDevice.IsReady) {
+        maybeRequestBatch(msg.device)
+    }
+
+    private suspend fun maybeRequestBatch(avoiding: Device? = null) {
+        val shuffledDevices = devices.values
```

Comment:
Can we use sequences? https://blog.kotlin-academy.com/effective-kotlin-use-sequence-for-bigger-collections-with-more-than-one-processing-step-649a15bb4bf